lib/rewards/server/subscribers/rewards-subscriber.js

```diff
@@ -0,0 +1,135 @@
+const { sc, Logger } = require('@reldens/utils');
+
+class RewardsSubscriber
+{
+    static async onBattleEnded(playerSchema, targetObject, events)
+    {
+        let rewards = sc.get(targetObject, 'rewards', false);
+
+        let winningRewards = RewardsSubscriber.getWinningRewards(rewards, false);
+        if (0 === winningRewards.length) {
+            return false;
+        }
+        let itemRepository =  targetObject.dataServer.getEntity('item')
+
+        for (let winningReward of winningRewards) {
+            let winningItem = await itemRepository.loadById(winningReward.item_id);
+            let addItemsResult = await RewardsSubscriber.addItemToPlayerInventory(playerSchema, winningItem, winningReward.drop_quantity);
+            if (!addItemsResult) {
+                Logger.error('Couldn\'t add item to Player\'s Inventory');
+                return false;
+            }
+
+            await events.emit('reldens.rewardDropped', {
+                reward: winningReward,
+                item_name: winningItem.label,
+                player_id: playerSchema.player_id,
+                room_id: targetObject.room_id,
+                client: playerSchema?.skillsServer?.client?.client
+            });
+        }
+        await RewardsSubscriber.updateUniqueRewards(winningRewards, targetObject.dataServer);
+        return true;
+    }
+
+    static async addItemToPlayerInventory(playerSchema, item, quantity)
+    {
+        let rewardItemInstance = playerSchema.inventory.manager.createItemInstance(item.key);
+        let rewardItems = Array(quantity).fill(rewardItemInstance);
```

Comment:
actually, the item object has it's own type to know if it's sigle instance or multiple instances, when you use the createItemInstance you can pass the quantity as second parameter and it will automatically return an array with the  amount of instances that should added

you can check htat in the reldens-items/lib/manager.js

remove the Array(quantity).fill()

Reply:
Nice! gonna change it.